operator_ui/install.go

```diff
@@ -0,0 +1,167 @@
+package main
+
+import (
+	"archive/tar"
+	"compress/gzip"
+	"context"
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"os"
+	"path"
+	"path/filepath"
+	"strings"
+	"time"
+)
+
+func main() {
+	const (
+		owner                  = "smartcontractkit"
+		repo                   = "operator-ui"
+		fullRepo               = owner + "/" + repo
+		tagPath                = "operator_ui/TAG"
+		unpackDir              = "core/web/assets"
+		downloadTimeoutSeconds = 10
+	)
+	rootDir := rootDir()
+
+	tag := mustReadTagFile(path.Join(rootDir, tagPath))
+	strippedTag := stripVersionFromTag(tag)
+	assetName := fmt.Sprintf("%s-%s-%s.tgz", owner, repo, strippedTag)
+	downloadUrl := fmt.Sprintf("https://github.com/%s/releases/download/%s/%s", fullRepo, tag, assetName)
+
+	// Assuming that we're in "root/operator_ui/"
+	unpackPath := filepath.Join(rootDir, unpackDir)
+	err := rmrf(unpackPath)
+	if err != nil {
+		log.Fatalln(err)
+	}
+
+	subPath := "package/artifacts/"
+	mustDownloadSubAsset(downloadUrl, downloadTimeoutSeconds, unpackPath, subPath)
+}
+
+func rootDir() string {
+	thisDir := thisExecutablesDir()
+
+	return filepath.Join(thisDir, "..")
+}
+
+func thisExecutablesDir() string {
+	exec, err := os.Executable()
+	if err != nil {
+		log.Fatalln(err)
+	}
+	dir := filepath.Dir(exec)
+
+	return dir
```

Comment:
Isn't this the source of the problem that you are trying to work around?

Reply:
In relation to the original shell script? That script was path resolution was broken overall, this relicates the intention of the shell scripts path resolution though

Reply:
It just seems like we have an odd workaround for the purpose of addressing this additional oddity, which is something we are choosing to do, not necessarily stuck having to do. It's not obvious to me why it can't be done cleanly. I am sympathetic to the idea that users could misuse the command by biffing something if they had to pass it in, but we are committing the full command to the repo for them to invoke indirectly, so users don't actually have to make this decision.